src/main/java/com/hivemq/persistence/PersistenceEntry.java

```diff
@@ -58,12 +58,11 @@ public int getEstimatedSize() {
             return sizeInMemory;
         }
 
-        int size = ObjectMemoryEstimation.objectOverheadSize(); // object overhead
+        int size = ObjectMemoryEstimation.objectShellSize(); // object overhead
         size += ObjectMemoryEstimation.longSize(); // timestamp
         size += ObjectMemoryEstimation.intSize(); // sizeInMemory
 
         // contained object
-        size += ObjectMemoryEstimation.objectOverheadSize();
```

Comment:
Imho this would then also need a objectRefSize for the reference from the collection